example/gosumtype_1_declaration.boilerplate.go

```diff
@@ -126,3 +199,89 @@ func UserMap[A any](value User, variants UserVariantsMap[A]) A {
 	})
 	return result
 }
+
+// User = Anonymous | Member | Admin
+type User struct {
+	user user
+}
+
+// user is the interface for UserVariants
+type user interface {
+	Match(variants UserVariants)
+}
+
+func (s User) Match(variants UserVariants) {
+	s.user.Match(variants)
+}
```

Comment:
[underneath]

because our sum type is implemented as a `struct` now
1. we need it to hold a value that could be one of the variant type, so we need an unexported interface type
2. also, we proxy `Match` method call so application code need not change (other than _removing_ their interface definition)